app/models/generic_object.rb

```diff
@@ -35,30 +33,35 @@ def inspect
     "#<#{self.class} #{attributes_as_string.join(", ")}>"
   end
 
-  def method_missing(method_name, *args)
-    m = method_name.to_s.chomp("=")
-    super if generic_object_definition && !generic_object_definition.defined_attributes.include?(m)
-    method_name.to_s.end_with?('=') ? custom_attribute_setter(m, args.first) : custom_attribute_getter(m)
-  end
-
-  def respond_to_missing?(method_name, *args)
-    return true unless generic_object_definition
-    generic_object_definition.defined_attributes.include?(method_name.to_s.chomp!("=")) || super
-  end
-
   def property_attributes
-    custom_attributes.each_with_object({}) { |ca, h| h[ca.name] = custom_attribute_getter(ca.name) if ca.id }
+    custom_attributes.each_with_object({}) { |ca, h| h[ca.name] = custom_attribute_getter(ca.name) }
   end
 
   def property_attributes=(options)
+    raise "generic_object_definition is nil" unless generic_object_definition
+    options.keys.each do |k|
+      unless generic_object_definition.property_attribute_defined?(k)
+        raise ActiveModel::UnknownAttributeError.new(self, k)
+      end
+    end
     options.each { |k, v| custom_attribute_setter(k, v) }
   end
 
   private
 
+  def method_missing(method_name, *args)
+    m = method_name.to_s.chomp("=")
+    super unless generic_object_definition.try(:property_attribute_defined?, m)
```

Comment:
Oh I see, if the generic_object_definition is missing, treat it like an undefined property attribute, which is true.  Clever!  I like it!